internal/authz/authz.go

```diff
@@ -0,0 +1,237 @@
+package authz
+
+import (
+	"context"
+	"fmt"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/openfga/openfga/pkg/authcontext"
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+const (
+	// API methods.
+	ReadAuthorizationModel  = "ReadAuthorizationModel"
+	ReadAuthorizationModels = "ReadAuthorizationModels"
+	Read                    = "Read"
+	Write                   = "Write"
+	ListObjects             = "ListObjects"
+	StreamedListObjects     = "StreamedListObjects"
+	Check                   = "Check"
+	ListUsers               = "ListUsers"
+	WriteAssertions         = "WriteAssertions"
+	ReadAssertions          = "ReadAssertions"
+	WriteAuthorizationModel = "WriteAuthorizationModel"
+	ListStores              = "ListStores"
+	CreateStore             = "CreateStore"
+	GetStore                = "GetStore"
+	DeleteStore             = "DeleteStore"
+	Expand                  = "Expand"
+	ReadChanges             = "ReadChanges"
+
+	// Relations.
+	CanCallReadAuthorizationModels  = "can_call_read_authorization_models"
+	CanCallRead                     = "can_call_read"
+	CanCallWrite                    = "can_call_write"
+	CanCallListObjects              = "can_call_list_objects"
+	CanCallCheck                    = "can_call_check"
+	CanCallListUsers                = "can_call_list_users"
+	CanCallWriteAssertions          = "can_call_write_assertions"
+	CanCallReadAssertions           = "can_call_read_assertions"
+	CanCallWriteAuthorizationModels = "can_call_write_authorization_models"
+	CanCallListStores               = "can_call_list_stores"
+	CanCallCreateStore              = "can_call_create_stores"
+	CanCallGetStore                 = "can_call_get_store"
+	CanCallDeleteStore              = "can_call_delete_store"
+	CanCallExpand                   = "can_call_expand"
+	CanCallReadChanges              = "can_call_read_changes"
+
+	StoreType       = "store"
+	ApplicationType = "application"
+	SystemType      = "system"
+	RootSystemID    = "fga"
+)
+
+var (
+	ErrorResponse = &openfgav1.ForbiddenResponse{Code: 403, Message: "the principal is not authorized to perform the action"}
+)
+
+type Config struct {
+	StoreID string
+	ModelID string
+}
+
+type Authorizer struct {
+	config *Config
+	server ServerInterface
+	logger logger.Logger
+}
+
+// NewAuthorizer creates a new authorizer.
```

Comment:
Could we make this comment more informative? Maybe use this to document decisions for future us, e.g. why this is not a middleware?

Reply:
Is this function really useful? It seems only to serve to populate the internal struct properties with values that are provided directly as function input.

In this case, it likely makes more sense to export the 3 struct properties (make them public) and remove this function entirely.

Also, as opposed to removing the comment, I would prefer to see an improved comment. If the function does nothing significant to comment on, that is a good indicator that the function isn't useful.

Reply:
@senojj this seems be be the pattern we use in the server: e.g. https://github.com/openfga/openfga/blob/main/pkg/logger/logger.go#L32-L37

If we want to change that pattern can we change that in a different set of work? (and can we make sure the team is on board?)

Reply:
I just want us, as a team, to be intentional with our code and not follow patterns blindly. Doing so would lead us to cargo cult thinking.

Reply:
We definitely have lots of comments in the codebase that serve no purpose but increase line count. I'd agree that just because that is the case we don't have to make the problem worse 😅